src/components/ItemsList.js

```diff
@@ -43,41 +47,62 @@ const ItemsList = () => {
   };
 
+  const clearText = () => setSearchTerm('');
 
   return (
     <div className="items-list">
       <h1>Your Shopping List</h1>
       {loading && <p>Loading...</p>}
       {error && <p>An error has occured...</p>}
       {shoppingList && !shoppingList.length && (
-            <div className="add-item">
-              <p>You haven't created a shopping list yet...</p>
-              <button type="submit" onClick={handleRedirect}>
-                Add First Item
-              </button>
-            </div>
-          )}
-      <form>
+        <div className="add-item">
+          <p>You haven't created a shopping list yet...</p>
+          <button type="submit" onClick={handleRedirect}>
+            Add First Item
+          </button>
+        </div>
+      )}
+      <form onReset={clearText} className="search-list">
+        {shoppingList && shoppingList[0] ? (
+          <div>
+            <input
+              type="text"
+              placeholder="Search list.."
+              value={searchTerm}
+              onChange={handleChange}
+            />
+            <input type="reset" value="X" />
+          </div>
+        ) : null}
         <ul>
           {shoppingList &&
             shoppingList[0] &&
-            shoppingList[0].items.map((shoppingItemObject, index) => {
-              return (
-                <li key={shoppingItemObject.shoppingListItemName + index}>
-                  <input
-                    type="checkbox"
-                    id={shoppingItemObject.shoppingListItemName}
-                    onChange={() => markItemAsPurchased(index)}
-                    checked={wasItemPurchasedWithinLastOneDay(
-                      shoppingItemObject.lastPurchasedOn,
-                    )}
-                  />
-                  <label htmlFor={shoppingItemObject.shoppingListItemName}>
-                    {shoppingItemObject.shoppingListItemName}
-                  </label>
-                </li>
-              );
-            })}
+            shoppingList[0].items
+              .filter((shoppingItemObject) =>
+                shoppingItemObject.shoppingListItemName
+                  .toLowerCase()
+                  .includes(searchTerm.toLowerCase()),
+              )
+              .map((shoppingItemObject, index) => {
+                const shopIndex = shoppingList[0].items.indexOf(
+                  shoppingItemObject,
+                );
+                return (
+                  <li key={shoppingItemObject.shoppingListItemName + index}>
+                    <input
+                      type="checkbox"
+                      id={shoppingItemObject.shoppingListItemName}
+                      onChange={() => markItemAsPurchased(shopIndex)}
+                      checked={wasItemPurchasedWithinLastOneDay(
+                        shoppingItemObject.lastPurchasedOn,
+                      )}
+                    />
+                    <label htmlFor={shoppingItemObject.shoppingListItemName}>
```

Comment:
This is to both teams. I think it is high time we considered refactoring this component and `AddItems` component render functionality so that they are left to become container components holding app logic and state. They  have become quite large and difficult to read.

Reply:
I agree. That is a great idea 👍